### PR TITLE
Update CI to Python 3.9

### PR DESCRIPTION
**Pull request by jgraham**
This uses 3.6 and 3.9 as the min/max supported configuration rather than 3.6/3.8

### CODE REVIEW

**Comment by foolip** [comment]
I've pushed a bunch of changes and would be shocked if I got them all right, @jgraham please review for things that our CI don't actually test that need manual confirmation.

File: tools/tox.ini [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [tox]
-envlist = py36,py37,py38,{py36,py37,py38}-flake8,{py36,py38}-mypy3
+envlist = py36,py37,py38,py39,{py36,py37,py38,py39}-flake8,{py36,py39}-mypy3
```
Comment (foolip) [inline tools/tox.ini:2]:
```suggestion
envlist = py36,py37,py38,py39,{py36,py37,py38,py39}-flake8,{py36,py37,py38,py39}-mypy3
```

Maybe? I'm often not sure if the envlist in `tox.ini` is everything that will be tested in CI, or everything that's *possible* to test. Certainly in some cases only a subset it tested, but are there any places where we test all combinations and shouldn't sprinkle more combinations than necessary?

Comment (jgraham) [reply tools/tox.ini]: In pratice I don't think we care about whether mypy in particular passes in intermediate versions of Python, but sure I guess this is reasonable.

Comment (foolip) [reply tools/tox.ini]: I pushed that change, mostly so I could test py38-mypy3 locally, since I don't have 3.9 installed.

**Comment by foolip** [comment]
The remaining problem now is this:
```
py36-mypy3 create: /home/test/web-platform-tests/tools/.tox/py36-mypy3
py36-mypy3 installdeps: -rrequirements_mypy.txt
py36-mypy3 installed: mypy==0.812,mypy-extensions==0.4.3,typed-ast==1.4.3,typing-extensions==3.7.4.3
py36-mypy3 run-test-pre: PYTHONHASHSEED='1082800144'
py36-mypy3 run-test: commands[0] | mypy --config-file=/home/test/web-platform-tests/tools/mypy.ini --no-incremental -p tools.manifest -p tools.lint -p tools.gitignore
tools/lint/lint.py:847: error: "object" has no attribute "__enter__"
tools/lint/lint.py:847: error: "object" has no attribute "__exit__"
Found 2 errors in 1 file (checked 32 source files)
```

I'm having trouble running this locally and don't know how to fix this.

**Comment by jgraham** [comment]
> The remaining problem now is this:
> 
> ```
> py36-mypy3 create: /home/test/web-platform-tests/tools/.tox/py36-mypy3
> py36-mypy3 installdeps: -rrequirements_mypy.txt
> py36-mypy3 installed: mypy==0.812,mypy-extensions==0.4.3,typed-ast==1.4.3,typing-extensions==3.7.4.3
> py36-mypy3 run-test-pre: PYTHONHASHSEED='1082800144'
> py36-mypy3 run-test: commands[0] | mypy --config-file=/home/test/web-platform-tests/tools/mypy.ini --no-incremental -p tools.manifest -p tools.lint -p tools.gitignore
> tools/lint/lint.py:847: error: "object" has no attribute "__enter__"
> tools/lint/lint.py:847: error: "object" has no attribute "__exit__"
> Found 2 errors in 1 file (checked 32 source files)
> ```
> 
> I'm having trouble running this locally and don't know how to fix this.

In theory `cd tools; tox -e py36-mypy`. The problem is that the `nullcontext` implementation for Python3.6 isn't recognised as a ContextManager.  I tried fixing this locally but it didn't work well. Maybe @gsnedders has an idea.

**Comment by foolip** [comment]
I have tried `tox -e py36-mypy` on a machine with a `python3.6` binary, but to no avail.

Since both flake8 and mypy are linters of sorts, is there really any point in ensuring they also work with 3.6, or should we just stop running them and ignore the problem?

**Comment by foolip** [comment]
Yeah, since mypy is clever enough to understand the `sys.version_info >= (3, 7)` condition, there does seem to be some value in running it on 3.6. I just found https://mypy.readthedocs.io/en/stable/common_issues.html#spurious-errors-and-locally-silencing-the-checker and `type: ignore` seems like an option ultimately.

**Review by foolip — APPROVED** [review]
I'm OK with landing with the `tools/lint/lint.py` hack if none of us can figure out a better way today. If so it would be nice to file an issue about it though.

**Review by foolip — APPROVED** [review]
This works now, remaining failures are https://github.com/web-platform-tests/wpt/issues/28745

Maybe best to land on Monday though

**Comment by foolip** [comment]
It's Monday!

**Pull request merged**